tests/e2e/vmlistener_test.go

```diff
@@ -333,6 +339,176 @@ func (s *VMListenerTestParams) TestFailoverAcrossVmOnVsan(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Attach the volume
+	out, err = dockercli.AttachVolume(s.vm1, s.volumeName, s.containerName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 1. Restart hostd
+	out, err = ssh.InvokeCommand(s.esx, restartHostd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Hostd reports running status in about a second but
+	// takes around 10 - 15 seconds to be able to serve requests
+	misc.SleepForSec(15)
+
+	// 2. Verify volume stays attached
+	status := verification.VerifyAttachedStatus(s.volumeName, s.vm1, s.esx)
+	c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
+
+	// 3. Kill this VM
+	isVMKilled := killVM(s.esx, s.vm1Name)
+	c.Assert(isVMKilled, Equals, true, Commentf("VM [%s] was not killed successfully", s.vm1Name))
+
+	// Give some time for the VM event to be handled
+	misc.SleepForSec(5)
+
+	// 4. Status should be detached
+	status = verification.VerifyDetachedStatus(s.volumeName, s.vm2, s.esx)
```

Comment:
Oh, yes its got to be polled, thats why I wanted the same name so test code doesn;t have to change.